src/dcp.jl

```diff
@@ -20,10 +20,19 @@ struct AffineVexity <: Vexity end
 struct ConvexVexity <: Vexity end
 struct ConcaveVexity <: Vexity end
 
+struct DCPViolationError <: Exception end
+
+function Base.showerror(io::IO, ::DCPViolationError)
+    return print(
+        io,
+        "DCPViolationError: Expression not DCP compliant. Trying to solve non-DCP compliant problems can lead to unexpected behavior.",
```

Comment:
So. Based on https://ask.cvxr.com, we need this error to be veeeeeery explicit. So many people ask questions. I'l make a change